cbr-service-profile/main.tf

```diff
@@ -48,6 +48,7 @@ locals {
           ref = {
             account_id   = data.ibm_iam_account_settings.iam_account_settings.account_id
             service_name = serviceref
+            location     = (serviceref == "compliance" || serviceref == "directlink" || serviceref == "iam-groups" || serviceref == "containers-kubernetes" || serviceref == "user-management") ? null : var.location
```

Comment:
I think this should be a validation block to return an explicit error message to the caller if location is set for a service not supporting location.

The current logic would silently ignore the location field which may make call think that there is a restriction per location when in fact there is not, which is really dangerous from a security perspective.